src/main/java/com/cloudogu/scm/repositorytemplate/RepositoryTemplater.java

```diff
@@ -0,0 +1,187 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.cloudogu.scm.repositorytemplate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.CustomClassLoaderConstructor;
+import org.yaml.snakeyaml.error.YAMLException;
+import sonia.scm.ContextEntry;
+import sonia.scm.NotFoundException;
+import sonia.scm.repository.BrowserResult;
+import sonia.scm.repository.FileObject;
+import sonia.scm.repository.InternalRepositoryException;
+import sonia.scm.repository.Repository;
+import sonia.scm.repository.RepositoryContentInitializer.InitializerContext;
+import sonia.scm.repository.api.RepositoryService;
+import sonia.scm.repository.api.RepositoryServiceFactory;
+import sonia.scm.template.Template;
+import sonia.scm.template.TemplateEngine;
+import sonia.scm.template.TemplateEngineFactory;
+
+import javax.inject.Inject;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Optional;
+
+public class RepositoryTemplater {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RepositoryTemplater.class);
+
+  private final TemplateEngineFactory engineFactory;
+  private final RepositoryServiceFactory repositoryServiceFactory;
+
+  // SnakeYaml cannot find the classes without the right classloader
+  private final Yaml yaml = new Yaml(new CustomClassLoaderConstructor(RepositoryTemplateCollector.class.getClassLoader()));
+  private InitializerContext context;
+
+  @Inject
+  public RepositoryTemplater(TemplateEngineFactory engineFactory, RepositoryServiceFactory repositoryServiceFactory) {
+    this.engineFactory = engineFactory;
+    this.repositoryServiceFactory = repositoryServiceFactory;
+  }
+
+  public void render(Repository template, InitializerContext context) {
+    this.context = context;
+    try (
+      RepositoryService templateService = repositoryServiceFactory.create(template);
+    ) {
+      render(templateService);
+    } catch (IOException e) {
+      LOG.error("could not use template on new repository", e);
+    }
+  }
+
+  private void render(RepositoryService templateService) throws IOException {
+    RepositoryTemplate repositoryTemplate = parseTemplateConfig(templateService);
+    TemplateEngine engine = setupTemplateEngine(repositoryTemplate);
+    TemplateFilter templateFilter = new TemplateFilter(context, engine, templateService.getRepository());
+
+    for (RepositoryTemplateFile templateFile : repositoryTemplate.getFiles()) {
+      BrowserResult result = templateService.getBrowseCommand().setPath(templateFile.getName()).setRecursive(true).getBrowserResult();
```

Comment:
I think we should cut of the first of the template file name, if the name starts with a "/".